db/internal_stats.h

```diff
@@ -237,6 +237,28 @@ class InternalStats {
       }
     }
 
+    CompactionStats& operator=(const CompactionStats& c) {
+      micros = c.micros;
+      cpu_micros = c.cpu_micros;
+      bytes_read_non_output_levels = c.bytes_read_non_output_levels;
+      bytes_read_output_level = c.bytes_read_output_level;
+      bytes_written = c.bytes_written;
+      bytes_moved = c.bytes_moved;
+      num_input_files_in_non_output_levels =
+          c.num_input_files_in_non_output_levels;
+      num_input_files_in_output_level = c.num_input_files_in_output_level;
+      num_output_files = c.num_output_files;
+      num_input_records = c.num_input_records;
+      num_dropped_records = c.num_dropped_records;
+      count = c.count;
+
+      int num_of_reasons = static_cast<int>(CompactionReason::kNumOfReasons);
+      for (int i = 0; i < num_of_reasons; i++) {
+        counts[i] = c.counts[i];
+      }
+      return *this;
+    }
```

Comment:
Pretty sure that they could have either avoided the duplication here or in the copy constructor. I've forgotten the syntax for doing so, but because this is already upstream it doesn't matter too much.

Reply:
I'm not sure why they didn't just say `CompactionStats& operator=(const CompactionStats& c) = default;`. But then again, I don't know these things any more.

Reply:
`counts` requires a deep copy which wouldn't happen with the default constructor.

To avoid duplication the copy constructor should call the assignment operator like `*this = other;`.